theano/gof/destroyhandler.py

```diff
@@ -394,6 +394,32 @@ def get_destroyers_of(r):
                 return []
         fgraph.destroyers = get_destroyers_of
 
+        def has_destroyers(protected_list):
```

Comment:
I think I would do something like this at the start of this method:

```
if not self.stale_droot:
    for var in protected_var:
        if self.destroyers(var):
            raise ...
```

This do 2 things:
1) We should always call has_destroyers what ever the value of cycle_detection is.
2) This could speed up or slow down the case when cycle_detection is the default value. So we need to speed test that. If is don't speed up computation, we should change the if to be `if self.cycle_detection != 'fast':`

What do you think of that?